src/main/kotlin/cc/hyperium/processes/services/commands/engine/DefaultArgumentParsers.kt

```diff
@@ -31,49 +27,62 @@ class StringArgumentParser : ArgumentParser {
             return arguments.poll()
         }
 
-        val greedy = param.annotations.firstOrNull { it.annotationClass == Greedy::class } as? Greedy
-        val take = param.annotations.firstOrNull { it.annotationClass == Take::class } as? Take
-        val quotable = param.annotations.firstOrNull { it.annotationClass == Quotable::class } as? Quotable
-
         val sb = StringBuilder()
         val wordOne = arguments.poll() ?: return null
 
-        if (greedy != null) {
-            sb.append(wordOne)
+        return when (val ann = param.annotations.first()) {
+            is Greedy -> greedy(sb, wordOne, arguments)
+            is Take -> take(sb, wordOne, arguments, ann)
+            is Quotable -> quotable(sb, wordOne, arguments)
+            else -> throw UnsupportedOperationException("Unknown annotation $ann")
+        }
+    }
 
-            while (arguments.peek() != null) {
-                sb.append(" ${arguments.poll()}")
-            }
-        } else if (take != null) {
-            var i = 0
+    private fun greedy(sb: StringBuilder, wordOne: String, arguments: ArgumentQueue): String {
+        sb.append(wordOne)
+
+        while (arguments.peek() != null) {
+            sb.append(" ${arguments.poll()}")
+        }
+
+        return sb.toString()
+    }
+
+    private fun take(sb: StringBuilder, wordOne: String, arguments: ArgumentQueue, take: Take): String {
+        sb.append(wordOne)
+
+        var i = 0
+
+        while (arguments.peek() != null && i < take.number) {
```

Comment:
We can emit the i++ line if we just put the i++ in the while statement.

Reply:
Of course, but I prefer to not write code like that because I feel like it makes it harder to understand.

Reply:
I would argue, its clear when you are using a incrementing loop for example the classic while(int i = 0; i>x; i++) is quiet conventional for most languages without the enhanced for loop.